Clients/Xamarin.Interactive.Client.Web/ClientApp/WebAssemblyAgentContainer.ts

```diff
@@ -0,0 +1,146 @@
+import { CodeCellEventType, ICodeCellEvent } from "./evaluation";
+import * as catalog from './i18n'
+import { StatusUIAction, MessageKind, MessageSeverity } from "./messages";
+import { WorkbookSession } from "./WorkbookSession";
+
+export class WebAssemblyAgentContainer
+{
+    private _wasmFrame: HTMLIFrameElement
```

Comment:
Two things. It took me a second to realize why you were prefixing with `_`.

* In this case, I think you can just make the fields `readonly` since they're both assigned in `constructor`:

    ```typescript
    readonly wasmFrame: HTMLIFrameElement
    readonly wasmMesssageChannel: MessageChannel
    ```

* If however you cannot (and question why, as we should try to be as immutable as possible), keep the field and the property together:

    ```typescript
    private _wasmMessageChannel: MessageChannel
    get wasmMessageChannel(): MessageChannel {
        return this._wasmMessageChannel
    }
    ```

Reply:
Yep, you're totally right. I forgot about `readonly` semantics in TS/JS. I did realize that `wasmFrameLoaded` does actually need to be a private setter, so it uses the suggested pattern of private field immediately followed by its property. :)